### PR TITLE
Update templates for function forms

### PR DESCRIPTION
**Pull request by leobalter**
While I was trying to cleanup the templates to work on #988, I ended up doing this all here (and not addressing #988 yet).

The fix for #988 will in another PR.

This adds a lot of missing templates for async functions, normalize files names, adds async generator templates for the arguments tests, and other smaller cleanups.

### CODE REVIEW

**Comment by leobalter** [comment]
For self-sanity check:

```
$ time test262-harness --hostType=jsshell -t 4 --hostPath=`which js`  $(git diff master --diff-filter=d --name-only test/)
Ran 944 tests
944 passed
0 failed

real	0m17.792s
user	0m41.292s
sys	0m13.016s
```

```
$ time test262-harness --hostType=d8 -t 4 --hostPath=`which d8` --hostArgs="--harmony-async-iteration" $(git diff master --diff-filter=d --name-only test/)
FAIL test/language/expressions/arrow-function/dflt-params-duplicates.js
  Expected test to throw error matching [object Object], but did not throw error

FAIL test/language/expressions/arrow-function/dflt-params-duplicates.js
  Expected test to throw error matching [object Object], but did not throw error

FAIL test/language/expressions/async-arrow-function/dflt-params-duplicates.js
  Expected test to throw error matching [object Object], but did not throw error

FAIL test/language/expressions/async-arrow-function/dflt-params-duplicates.js
  Expected test to throw error matching [object Object], but did not throw error

FAIL test/language/expressions/async-arrow-function/dflt-params-rest.js
  Expected test to throw error matching [object Object], but did not throw error

FAIL test/language/expressions/async-arrow-function/dflt-params-rest.js
  Expected test to throw error matching [object Object], but did not throw error

FAIL test/language/expressions/async-arrow-function/rest-params-trailing-comma-early-error.js
  Expected test to throw error matching [object Object], but did not throw error

FAIL test/language/expressions/async-arrow-function/rest-params-trailing-comma-early-error.js
  Expected test to throw error matching [object Object], but did not throw error

Ran 944 tests
936 passed
8 failed

real	0m17.620s
user	0m9.173s
sys	0m30.678s
```

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Place your settings in this file to overwrite default and user settings.
+{
+	  "editor.tabSize": 2,
```
Comment (rwaldron) [inline .vscode/settings.json]: lol

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    "editor.detectIndentation": false,
```
Comment (rwaldron) [inline .vscode/settings.json]: Trailing commas aren't allowed in JSON

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+}
```
Comment (rwaldron) [inline .vscode/settings.json]: Also... did you mean to commit this file?

File: src/arguments/default/cls-expr-async-gen-meth-static.template [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
```
Comment (rwaldron) [inline src/arguments/default/cls-expr-async-gen-meth-static.template]: Is this file just moving from another location, or is it new?

File: src/arguments/default/cls-expr-async-gen-meth-static.template [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+features: [async-iteration]
+---*/
+
+var callCount = 0;
+var C = class {
+  static async *method() {
+    /*{ body }*/
+    callCount = callCount + 1;
+  }
+}
```
Comment (rwaldron) [inline src/arguments/default/cls-expr-async-gen-meth-static.template]: ;

File: src/arguments/default/cls-expr-async-gen-meth.template [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+features: [async-iteration]
+---*/
+
+var callCount = 0;
+var C = class {
+  async *method() {
+    /*{ body }*/
+    callCount = callCount + 1;
+  }
+}
```
Comment (rwaldron) [inline src/arguments/default/cls-expr-async-gen-meth.template]: ;

File: src/arguments/default/cls-expr-meth.template [hunk 7]
```diff
@@ -15,7 +15,7 @@ info: |
 
 var callCount = 0;
 var C = class {
-  method(/*{ params }*/) {
```
Comment (rwaldron) [inline src/arguments/default/cls-expr-meth.template]: Did this form move to another template?

File: .vscode/settings.json (hunk 3 above)
Comment (leobalter) [inline .vscode/settings.json]: I can - and will - exclude this commit, I'm using vscode and it started messing with the indentation.

File: src/arguments/default/cls-expr-meth.template (hunk 7 above)
Comment (leobalter) [inline src/arguments/default/cls-expr-meth.template]: on the arguments folder, there's no case providing `/*{ params }*/` that's what I removed in there.

**Comment by rwaldron** [comment]
I've run the sanity checks locally, producing the same results.

**Comment by rwaldron** [comment]
- templates and cases look good to me. 
- the re-org makes sense


The only thing is a few minor nits, inline.

File: src/arguments/default/cls-expr-async-gen-meth-static.template (hunk 4 above)
Comment (leobalter) [inline src/arguments/default/cls-expr-async-gen-meth-static.template]: These are the same templates used in the src/function-forms/default folder (old src/params/default).

I'm copying them.

**Comment by leobalter** [comment]
Done.

To minimize the amount of commits, even if they are all doing different things, I squashed all the Generate tests into a single commit. 

I removed the vscode config commit as it doesn't belong here and fixed the nitpicks.

**Pull request merged**